pyramid/tierFS.go

```diff
@@ -0,0 +1,143 @@
+package pyramid
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/treeverse/lakefs/block"
+)
+
+type storageProps struct {
+	t                  StorageType
+	localDir           string
+	blockStoragePrefix string
+}
+
+// ImmutableTierFS is a filesystem where written files are never edited.
+// All files are stored in the block storage. Local paths are treated as a
+// cache layer that will be evicted according to the given eviction algorithm.
+type ImmutableTierFS struct {
+	adaptor block.Adapter
+	config  map[StorageType]storageProps
+
+	// TODO: use refs anc last-access for the eviction algorithm
```

Comment:
We have a `cache` package in this project that uses golang-lru and could probably be useful here.  However caches tend to have particular semantics, consider picking one cache and coding around that one from the start (to avoid surprises).

Reply:
Why do I need cache in here? I need to keep the access of all opened files, it doesn't make sense to drop any record here.